File: backend/app/schemas/attachment.py
from datetime import datetime
from typing import Optional
from uuid import UUID

from pydantic import BaseModel


class AttachmentBase(BaseModel):
    product_id: UUID
    field_key: str
    alias: Optional[str] = None
    drive_file_id: str
    file_name: str
    mime_type: Optional[str] = None
    url: Optional[str] = None


class AttachmentCreate(AttachmentBase):
    pass


class AttachmentUpdate(BaseModel):
    alias: Optional[str] = None
    url: Optional[str] = None


class AttachmentInDBBase(AttachmentBase):
    id: UUID
    uploaded_at: datetime

    class Config:
        orm_mode = True


class Attachment(AttachmentInDBBase):
    pass


class AttachmentInDB(AttachmentInDBBase):
    pass 

File: frontend/src/types/attachment.ts
export interface Attachment {
  id: string;
  product_id: string;
  field_key: string;
  alias?: string | null;
  drive_file_id: string;
  file_name: string;
  mime_type?: string | null;
  url?: string | null;
  uploaded_at: string;
}
